app/src/main/kotlin/com/wire/android/di/CoreLogicModule.kt

```diff
@@ -124,62 +126,60 @@ class UseCaseModule {
+        coreLogic.getSessionScope(currentAccount).conversations.getConversationDetails
 
     @ViewModelScoped
     @Provides
-    fun getMessagesUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentSession session: AuthSession) =
-        coreLogic.getSessionScope(session).messages.getRecentMessages
+    fun getMessagesUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: String) =
+        coreLogic.getSessionScope(currentAccount).messages.getRecentMessages
 
     @ViewModelScoped
     @Provides
-    fun deleteClientUseCase(@CurrentSession currentSession: AuthSession, clientScopeProviderFactory: ClientScopeProvider.Factory) =
-        clientScopeProviderFactory.create(currentSession).clientScope.deleteClient
+    fun deleteClientUseCase(@CurrentAccount currentAccount: String, clientScopeProviderFactory: ClientScopeProvider.Factory) =
+        clientScopeProviderFactory.create(currentAccount).clientScope.deleteClient
 
     @ViewModelScoped
     @Provides
-    fun registerClientUseCase(@CurrentSession currentSession: AuthSession, clientScopeProviderFactory: ClientScopeProvider.Factory) =
-        clientScopeProviderFactory.create(currentSession).clientScope.register
+    fun registerClientUseCase(@CurrentAccount currentAccount: String, clientScopeProviderFactory: ClientScopeProvider.Factory) =
+        clientScopeProviderFactory.create(currentAccount).clientScope.register
 
     @ViewModelScoped
     @Provides
-    fun listenToEventsUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentSession session: AuthSession) =
-        coreLogic.getSessionScope(session).listenToEvents
+    fun listenToEventsUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: String) =
+        coreLogic.getSessionScope(currentAccount).listenToEvents
 
     @ViewModelScoped
     @Provides
-    fun providesGetSelfUseCase(
-        @KaliumCoreLogic coreLogic: CoreLogic,
-        @CurrentSession currentSession: AuthSession
-    ): GetSelfUserUseCase =
-        coreLogic.getSessionScope(currentSession).users.getSelfUser
+    fun providesGetSelfUseCase(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: String): GetSelfUserUseCase =
+        coreLogic.getSessionScope(currentAccount).users.getSelfUser
 
     @ViewModelScoped
     @Provides
     fun providesSendTextMessageUseCase(
         @KaliumCoreLogic coreLogic: CoreLogic,
-        @CurrentSession currentSession: AuthSession
+        @CurrentAccount currentAccount: String
     ): SendTextMessageUseCase =
-        coreLogic.getSessionScope(currentSession).messages.sendTextMessage
+        coreLogic.getSessionScope(currentAccount).messages.sendTextMessage
+
+    @ViewModelScoped
+    @Provides
+    fun logoutUseCaseProvider(@KaliumCoreLogic coreLogic: CoreLogic, @CurrentAccount currentAccount: String): LogoutUseCase =
```

Comment:
As we are now passing the current account / user ID could this be renamed to currentAccountID? so it makes more sense, as for my understanding `currentAccount` would be an `Account` class and not only the ID as string 🤔 

If so, then the same for the parameters changed above

Reply:
We can do this in a later stage